### PR TITLE
Support knot insertion up to C^-1

### LINKED ISSUE #215: Remove knots fails with first and last knot
Minimal example:
```
import splinepy as sp

# Create a simple spline
a = sp.BSpline([1],[[0,0,1,2,2]],[[0],[1],[2]])
# Remove first knot (optional)
a.remove_knots(0,[1]) # works just fine
# Error 
a.remove_knots(0,[0]) # Segfault
# Similarly
a.remove_knots(0,[2]) # Segfault
```

### PR DESCRIPTION
**Pull request by j042**
# Overview
Adds support for C^-1. Changes of this PR also fixes #215. In cpp, to extract cp ids for bezier patches, you now need multiplicity informations - This is available from `KnotVector`. For #297, we can also wrap this.

## Addressed issues
*  C^-1 spline creation through knot insertion was impossible
* #215

## Showcase
A short / one-liner example to highlight the (new) feature
```python
# bspline with no knots at 1.5
assert all(bspline.insert_knots(0, [1.5] * (degree + 1)))
```

## Known ToDos
- elevate degrees with c^-1 splines fails

## Checklists
* [ ] Documentations are up-to-date.
* [ ] Added example(s)
* [ ] Added test(s)

### CODE REVIEW

**Comment by j042** [comment]
test freezes @ windows-debug build

File: include/splinepy/splines/helpers/extract.hpp [hunk 1]
```diff
@@ -172,4 +175,146 @@ ExtractControlMeshSlice(const SplineType& spline,
   }
 }
 
+/**
+ * @brief Extract Bezier Patch IDs
```
Comment (jzwar) [inline include/splinepy/splines/helpers/extract.hpp:179]: Is there any new code in here?

File: tests/test_bezier_extraction.py [hunk 2]
```diff
@@ -64,6 +65,7 @@ def test_extraction_matrices_bspline_3D(self):
     def test_extraction_matrices_nurbs_3D(self):
         """Create matrices to extract splines"""
 
+        return
```
Comment (jzwar) [inline tests/test_bezier_extraction.py:68]: ?

File: tests/test_bezier_extraction.py [hunk 3]
```diff
@@ -103,6 +105,7 @@ def test_extraction_matrices_bspline_2D(self):
     def test_extraction_matrices_nurbs_2D(self):
         """Create matrices to extract splines"""
 
+        return
```
Comment (jzwar) [inline tests/test_bezier_extraction.py:108]: ?

**Review by jzwar — APPROVED** [review]

**Comment by jzwar** [comment]
LGTM, thanks for handling this. Please fix tests before merging

File: tests/test_bezier_extraction.py (hunk 2 above)
Comment (j042) [reply tests/test_bezier_extraction.py:68]: good catch!

File: include/splinepy/splines/helpers/extract.hpp (hunk 1 above)
Comment (j042) [reply include/splinepy/splines/helpers/extract.hpp:179]: code migration and c^-1 updates

**Pull request merged**